test/math.jl

```diff
@@ -842,24 +842,103 @@ end
         @test atan2(-T(1235.2341234), -T(2.5)) === -(T(pi)-(atan(abs(-T(1235.2341234)/T(2.5)))-_ATAN2_PI_LO(T)))
     end
 end
+#prev, current, next float
+pcnfloat(x) = prevfloat(x), x, nextfloat(x)
+@testset "sinh" begin
+    for T in (Float32, Float64)
+        @test sinh(zero(T)) === zero(T)
+        @test sinh(-zero(T)) === -zero(T)
+        @test sinh(nextfloat(zero(T))) === nextfloat(zero(T))
+        @test sinh(prevfloat(zero(T))) === prevfloat(zero(T))
+        @test sinh(T(1000)) === T(Inf)
+        @test sinh(-T(1000)) === -T(Inf)
+        @test isnan_type(T, sinh(T(NaN)))
+    end
+    for x in (pcnfloat(2.0^-28)..., pcnfloat(22.0)..., pcnfloat(709.7822265633563)...)
+        @test sinh(x) ≈ sinh(big(x)) rtol=eps(Float64)
```

Comment:
Seems simpler to do `@test sinh(x) ≈ sinh(big(x)) ≈ -sinh(x) rtol=eps(T)`, but it looks like this syntax isn't supported by `@test`.  😢

Reply:
Yeah, you suggested that above and I tried it but without luck so I just split it out into two.